### ISSUE TITLE
1. Add info to error messages in stdout 2. Define what "the breakline" means 3. Add "do's" and "dont's" to configuration format page

### ISSUE THREAD

**Issue by pmeaney-sqrt** [fluent/fluent-bit#5800]
Howdy folks! Hope you're having a great week! I am configuring a basic fluent-bit container to test out its grep plugin and make sure I can configure it to do what I need.

I am having a problem regarding the format of my fleunt-bit.conf and that of a test log file.

I’m editing my .conf file and getting this error: `[2022/07/28 14:55:42] [error] [config] error in /etc/fluent-bit/lines-1.txt:1: undefined key - check config is in valid classic format`

**It would be helpful if the error output would notify me of the particular line of the file with an undefined key, and that the problem is in a particular file-- in this case, it must be in the .conf file.**

I also received this error recently: `[2022/07/28 14:44:05] [error] [config] error in /etc/fluent-bit/realistic-example.txt:1: could not allocate key value pair` regarding the log file shown below

**I'd love it if the stdout would tell me the particular line, or key/value pair, or file in question.**

I've reviewed the documentation on format & schema:
https://docs.fluentbit.io/manual/administration/configuring-fluent-bit/classic-mode/format-schema

For me personally, I don't think the page above is very clear about how the .conf files need to be formatted.  It would be good to know exactly what is expected, and in very clear terms.  I want to see examples of good & bad formats-- i.e. "do's" and "don'ts".  I want to know why my configuration is broken, but this documentation page is lacking examples.

- For example, the word "space" appears twice, and "tab" appears zero times.  So there is no discussion about spaces vs tabs.  I see 4 spaces is "ideal".  All good there-- looks like tabs break things though.  It would be good to know explicitly that I cannot use tabs.
- [Quoting the docs page "A key must contain a value which ends in the breakline."](https://docs.fluentbit.io/manual/administration/configuring-fluent-bit/classic-mode/format-schema)  What does "ends in the breakline" mean? What does "The Breakline" mean to you?  It might mean different things to different people.  And sometimes we use the word "The" to refer to one specific, special case of a thing.  I would really like some clarity there.  Does it mean a "carriage return" or a some sort of breakline symbol? What is "The Breakline"?  I'd like to see some examples of Do's and Don'ts.  Such as "When ending a line, just press enter.  Be sure there is no space remaining at the end of the line."
- On the concept of a "break line" --  a "Space" "breaks" a line of text.  Between the key and value of a configuration line, do I need just one space? Or do the values need to be aligned visually? (That's where tabs would be handy). for example, below, "tail" and "on" are not aligned.  I hope that's ok, but I don't know if it could be a problem.  or if a "break line" could somehow mean those values need to be aligned vertically, as shown in the second example
- Without any info from the error directing me to the problem in the configuration file... it makes it difficult to know the source of the error.

My confusion example 1
```bash
[INPUT]
    Name   tail
    Read_from_Head on
```

My confusion example 2 -- the "Line" of text is "broken" such that things are vertically aligned.
```bash
[INPUT]
    Name                  tail
    Read_from_Head        on
```

Basically, there's

Thanks!
<img width="545" alt="Screen Shot 2022-07-28 at 9 59 44 AM" src="https://user-images.githubusercontent.com/102547768/181572991-9f1089f4-1223-4e38-9a65-489ec42065d0.png">

<img width="682" alt="Screen Shot 2022-07-28 at 10 10 43 AM" src="https://user-images.githubusercontent.com/102547768/181573205-e9533acc-fcbe-4db7-8e80-c59b652bfc6c.png">

file: fluent-bit.conf

```bash
[SERVICE]
    flush        1
    daemon       Off
    log_level    info
    parsers_file parsers.conf
    plugins_file plugins.conf
    http_server  Off
    http_listen  0.0.0.0
    http_port    2020
    storage.metrics on

[INPUT]
    Name   tail
    Path   /etc/fluent-bit/lines-1.txt
    # Read from the beginning of the file
    Read_from_Head on

[OUTPUT]
    Name        stdout
    Match       *

```

file: lines-basic-grep.txt
```bash
{"container_name": "containerABC", "color": "purple"}
{"container_name": "containerXYZ", "color": "purple"}
{"container_name": "containerABC", "color": "purple"}
{"container_name": "containerABC", "color": "yellow"}
{"container_name": "containerXYZ", "color": "yellow"}
{"container_name": "containerXYZ", "color": "red"}
{"container_name": "containerABC", "color": "red"}
{"container_name": "containerABC", "color": "red"}
{"container_name": "containerABC", "color": "blue"}
{"container_name": "containerXYZ", "color": "blue"}
{"container_name": "containerABC", "color": "green"}
{"container_name": "containerXYZ", "color": "green"}
```

My goal-- eventually-- is to set up a "grep" filter section, similar to this:

```bash
    Name grep
    Match /containerABC/
    Exclude message /purple/ /red/

or:

[FILTER]
    Name grep
    Rule ^container_name$ /containerABC/
    Exclude message /purple/ /red/
```

**Comment by pmeaney-sqrt** [comment]
Chatting with Pat just now in the Meetup--
He showed me that I was probably pointing fleunt-bit to the log file instead of the config file.  Yep! oops!

I was doing this-- pointing to log file on accident instead of the conf file:

`fluent-bit -c /etc/fluent-bit/lines-1.txt`

instead of:
`fluent-bit -c /etc/fluent-bit/fluent-bit.conf`

So, now I see that I am missing the 'name' property in [OUTPUT]

```
[2022/07/28 16:42:43] [error] [config] error in /etc/fluent-bit/fluent-bit.conf:19: undefined key - check config is in valid classic format
[2022/07/28 16:42:43] [error] [config] section 'output' is missing the 'name' property
[2022/07/28 16:42:43] [error] configuration file contains errors, aborting.
```

**Comment by pmeaney-sqrt** [comment]
Closing this issue as I will 1. re-create it in fluent-bit-docs repo 2. break into individual issues/PRs
